Test failure on Linux (CI) potentially related to OpenCL driver version?

The current failure in the CI/linux mentions this issue:

```
self = <pyopencl._cl.Kernel object at 0x7fddfe3f4950>
queue = <pyopencl._cl.CommandQueue object at 0x7fde0a47fdd0>
global_size = (10, 1), local_size = None
arg0 = <pyopencl._cl.Buffer object at 0x7fddfe85def0>, arg1 = array([0])
arg2 = array([0.], dtype=float32), global_offset = None, g_times_l = None
allow_empty_ndrange = False, wait_for = None

>   ???
E   pyopencl._cl.LogicError: when processing argument #2 (1-based): clSetKernelArg failed: INVALID_ARG_SIZE
```

I suspect that changing [this line](https://github.com/clEsperanto/pyclesperanto_prototype/blob/master/pyclesperanto_prototype/_tier0/execute.py#L114):
```
arguments.append(np.array([value], np.float32))
```
 
to this line:
```
arguments.append(np.array([value]))
```

could fix the issue. Unfortunately, tests fail then on my Windows system. I'll try later this week on my Linux machine. Just wanted to raise the point: There might be two different strategies for fixing this:
* Build in a switch which executes either the one or the other line depending on operating system / driver / version / ....
* Change the driver used in the CI?

Cheers,
Robert

> I assume the tests also run through on your mac?

Everything is executing correctly on my mac, but these four tests fail the `np.allclose` test:
```
FAILED tests/test_nonzero_maximum_box.py::test_nonzero_maximum_box
FAILED tests/test_nonzero_maximum_diamond.py::test_nonzero_maximum_diamond
FAILED tests/test_nonzero_minimum_box.py::test_nonzero_minimum_box
FAILED tests/test_nonzero_minimum_diamond.py::test_nonzero_minimum_diamond
```

> And I added a new kernel, which caused issues as well.

Ah.  Ok well that's a "good" problem :) gotta add kernels!  Here's one strategy for "kicking the can down the road" a bit.  Because it can be so useful to have generally passing tests on CI (so that as you work on many different PRs you can quickly see if the current PR has broken any existing functionality), and because I definitely expect that we'll run into many platform-specific issues here... I think we should make liberal use of [`pytest.mark.xfail`](https://docs.pytest.org/en/latest/skipping.html), which lets you say "I expect this test to currently fail, in these conditions, with this specific exception":

See for instance what I did in [test_maximum_sphere](https://github.com/clEsperanto/pyclesperanto_prototype/blob/0fffcbfbc48b3b29334d6cff1c1173f524342c7d/tests/test_maximum_sphere.py):

https://github.com/clEsperanto/pyclesperanto_prototype/blob/0fffcbfbc48b3b29334d6cff1c1173f524342c7d/tests/test_maximum_sphere.py#L4-L10

The first string (`'LINUX and CI'`) specifies the conditions under which this test is "allowed" to fail.  So that you might have it working locally (`not CI`) but not on CI, but you're "ok" with that for now.  The `raises=cl.LogicError` part says exactly what exception you expect... and a nice trick is to use the [`strict=True` xfail option](https://docs.pytest.org/en/latest/skipping.html#strict-parameter), which will actually _fail_ the test if it "passes" (unexpectedly).  This lets you just run tests while trying to fix your code, and if you get a test failure it means you fixed it (at which point you remove the xfail mark).

So if you want to mark the 7 remaining tests that are failing, we can get our CI working again so that as we work on other features we can see if we've broken new things, or if it's just the "expected" bugs

--- a/tests/test_close_index_gaps_in_label_maps.py
+++ b/tests/test_close_index_gaps_in_label_maps.py
@@ -1,6 +1,11 @@
 import pyclesperanto_prototype as cle
 import numpy as np
+import pyopencl as cl
+import pytest
 
+from . import LINUX, CI
+
+@pytest.mark.xfail('LINUX and CI', reason='INVALID_ARG_SIZE on CI', raises=cl.LogicError)
 def test_close_index_gaps_in_label_maps():
 
     gpu_input = cle.push(np.asarray([

--- a/tests/test_connected_components_labeling_box.py
+++ b/tests/test_connected_components_labeling_box.py
@@ -1,6 +1,11 @@
 import pyclesperanto_prototype as cle
 import numpy as np
+import pyopencl as cl
+import pytest
 
+from . import LINUX, CI
+
+@pytest.mark.xfail('LINUX and CI', reason='INVALID_ARG_SIZE on CI', raises=cl.LogicError)
 def test_connected_components_labeling_box():
     
     gpu_input = cle.push(np.asarray([

--- a/tests/test_copy_slice.py
+++ b/tests/test_copy_slice.py
@@ -3,8 +3,9 @@ import numpy as np
 import pytest
 import pyopencl as cl
 
+from . import LINUX, CI
 
-@pytest.mark.xfail(raises=cl.RuntimeError)
+@pytest.mark.xfail('LINUX and CI', reason='INVALID_ARG_SIZE on CI', raises=cl.LogicError)
 def test_copy_slice_from_3d():
 
     test1 = cle.push(np.asarray([

--- a/tests/test_nonzero_maximum_box.py
+++ b/tests/test_nonzero_maximum_box.py
@@ -1,10 +1,6 @@
 import pyclesperanto_prototype as cle
 import numpy as np
-import pytest
-import pyopencl as cl
 
-
-@pytest.mark.xfail(raises=cl.RuntimeError)
 def test_nonzero_maximum_box():
     test = cle.push(np.asarray([
         [0, 0, 0, 0, 0],

--- a/tests/test_nonzero_maximum_box.py
+++ b/tests/test_nonzero_maximum_box.py
@@ -24,6 +24,10 @@ def test_nonzero_maximum_box():
 
     result = cle.create(test)
     flag = cle.create((1, 1, 1))
+
+    # as nonzero filters don't touch zero values, we need to initialize the result in advance
+    cle.set(result, 0);
+
     cle.nonzero_maximum_box(test, flag, result)
 
     print(result)

--- a/tests/test_nonzero_maximum_diamond.py
+++ b/tests/test_nonzero_maximum_diamond.py
@@ -1,10 +1,6 @@
 import pyclesperanto_prototype as cle
 import numpy as np
-import pytest
-import pyopencl as cl
 
-
-@pytest.mark.xfail(raises=cl.RuntimeError)
 def test_nonzero_maximum_diamond():
     test = cle.push(np.asarray([
         [0, 0, 0, 0, 0],

--- a/tests/test_nonzero_maximum_diamond.py
+++ b/tests/test_nonzero_maximum_diamond.py
@@ -24,6 +24,10 @@ def test_nonzero_maximum_diamond():
 
     result = cle.create(test)
     flag = cle.create((1, 1, 1))
+
+    # as nonzero filters don't touch zero values, we need to initialize the result in advance
+    cle.set(result, 0);
+
     cle.nonzero_maximum_diamond(test, flag, result)
 
     print(result)

--- a/tests/test_nonzero_minimum_box.py
+++ b/tests/test_nonzero_minimum_box.py
@@ -1,10 +1,6 @@
 import pyclesperanto_prototype as cle
 import numpy as np
-import pytest
-import pyopencl as cl
 
-
-@pytest.mark.xfail(raises=cl.RuntimeError)
 def test_nonzero_minimum_box():
     test = cle.push(np.asarray([
         [0, 0, 0, 0, 0],

--- a/tests/test_nonzero_minimum_box.py
+++ b/tests/test_nonzero_minimum_box.py
@@ -24,6 +24,10 @@ def test_nonzero_minimum_box():
 
     result = cle.create(test)
     flag = cle.create((1, 1, 1))
+
+    # as nonzero filters don't touch zero values, we need to initialize the result in advance
+    cle.set(result, 0);
+
     cle.nonzero_minimum_box(test, flag, result)
 
     print(result)

--- a/tests/test_nonzero_minimum_diamond.py
+++ b/tests/test_nonzero_minimum_diamond.py
@@ -24,6 +24,10 @@ def test_nonzero_minimum_diamond():
 
     result = cle.create(test)
     flag = cle.create((1, 1, 1))
+
+    # as nonzero filters don't touch zero values, we need to initialize the result in advance
+    cle.set(result, 0);
+
     cle.nonzero_minimum_diamond(test, flag, result)
 
     print(result)

--- a/tests/test_nonzero_minimum_diamond.py
+++ b/tests/test_nonzero_minimum_diamond.py
@@ -1,10 +1,6 @@
 import pyclesperanto_prototype as cle
 import numpy as np
-import pytest
-import pyopencl as cl
 
-
-@pytest.mark.xfail(raises=cl.RuntimeError)
 def test_nonzero_minimum_diamond():
     test = cle.push(np.asarray([
         [0, 0, 0, 0, 0],

--- a/tests/test_set_nonzero_pixels_to_pixelindex.py
+++ b/tests/test_set_nonzero_pixels_to_pixelindex.py
@@ -1,7 +1,11 @@
 import pyclesperanto_prototype as cle
 import numpy as np
+import pyopencl as cl
+import pytest
 
+from . import LINUX, CI
 
+@pytest.mark.xfail('LINUX and CI', reason='INVALID_ARG_SIZE on CI', raises=cl.LogicError)
 def test_set_nonzero_pixels_to_pixelindex():
     test1 = cle.push_zyx(np.asarray([
         [0, 0, 0, 1],

--- a/tests/test_set_plane.py
+++ b/tests/test_set_plane.py
@@ -3,8 +3,7 @@ import numpy as np
 import pytest
 import pyopencl as cl
 
-
-@pytest.mark.xfail(raises=cl.RuntimeError)
+@pytest.mark.xfail('LINUX and CI', reason='INVALID_ARG_SIZE on CI', raises=cl.LogicError)
 def test_set_plane():
     result = cle.push(np.asarray([
         [

